### PR TITLE
- add filter for firebase jwt token verification and authorization

### PR DESCRIPTION
**Pull request by Kusibab**
Add filter for firebase jwt token verification and authorization. It still do not have valid principals and authorities flow becouse of lack of business logic for that.

### CODE REVIEW

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import lombok.Builder;
+
+@Builder
+public class FirebaseFilter extends OncePerRequestFilter {
```
Comment (mukasz) [inline weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java:31]: You should write some unit tests of that class - it implements a critical application logic.

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    private static String AUTH_HEADER = "Authorization";
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String path = request.getRequestURI();
+        // TODO: Should be all public endpoints
+        if ("/auth/register".equals(path)) {
```
Comment (mukasz) [inline weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: Provide it in application.yml as a list

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+            logger.debug("Successfully Authenticated");
+        }
+        filterChain.doFilter(request, response);
+
+    }
+
+    private FirebaseToken verifyIdToken(String idToken) throws FirebaseAuthException {
+        if (StringUtils.isEmpty(idToken)) {
+            throw new IllegalArgumentException("Token is empty");
```
Comment (mukasz) [inline weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: It is not a suitable exception. I think, we should create custom exception(WeddineoUnauthorisedException or sth like that).
In future, after implementing GlobalExceptionHandler this exception should be translated to HTTP 401.

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        if (!StringUtils.isEmpty(authToken)) {
+            Authentication auth;
+            try {
+                auth = getAuthentication(authToken);
+                SecurityContextHolder.getContext().setAuthentication(auth);
+            } catch (FirebaseAuthException e) {
+                throw new WeddineoFirebaseRuntimeException("Cannot connect to firebase auth", e);
+            }
+
+            logger.debug("Successfully Authenticated");
```
Comment (mukasz) [inline weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: This log tells nothing in my opinion :D It should contains user Id, request id or sth that allows us to indetify operation.
But I think information about success is not obligatory - no exceptions thrown meas that everything is correct :D

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    private FirebaseToken verifyIdToken(String idToken) throws FirebaseAuthException {
+        if (StringUtils.isEmpty(idToken)) {
+            throw new IllegalArgumentException("Token is empty");
+        }
+        return FirebaseAuth.getInstance().verifyIdToken(idToken);
+    }
+
+    private Authentication getAuthentication(String idToken) throws FirebaseAuthException {
+        FirebaseToken token = verifyIdToken(idToken);
+        assert token != null;
```
Comment (mukasz) [inline weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: I wolud prefer `Optional.ofNullable().orElseThrow()` - with that construction you can throw excepetion that describes problem in better. Now AssertationError will be thrown here.

**Review by mukasz — CHANGES_REQUESTED** [review]

**Review by PrzemyslawZnojek — APPROVED** [review]

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java (hunk 3 above)
Comment (Kusibab) [reply weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: done

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java (hunk 4 above)
Comment (Kusibab) [reply weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: done

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java (hunk 5 above)
Comment (Kusibab) [reply weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: done

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java (hunk 2 above)
Comment (Kusibab) [reply weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java]: done

File: weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java (hunk 1 above)
Comment (Kusibab) [reply weddineo-core/src/main/java/com/weddineo/configuration/filter/FirebaseFilter.java:31]: will be in the future

**Review by mukasz — APPROVED** [review]

**Pull request merged**